sdk/src/main/java/io/radar/sdk/RadarVerificationManager.kt

```diff
@@ -40,267 +40,328 @@ internal class RadarVerificationManager(
-                                                uids,
-                                                false,
-                                                object : Radar.RadarBeaconCallback {
-                                                    override fun onComplete(
-                                                        status: Radar.RadarStatus,
-                                                        beacons: Array<RadarBeacon>?
-                                                    ) {
-                                                        if (status != Radar.RadarStatus.SUCCESS || beacons == null) {
-                                                            callTrackApi(null)
-
-                                                            return
-                                                        }
-
-                                                        callTrackApi(beacons)
-                                                    }
-                                                })
-                                        } else if (beacons != null) {
-                                            Radar.beaconManager.startMonitoringBeacons(beacons)
-
-                                            Radar.beaconManager.rangeBeacons(
-                                                beacons,
-                                                false,
-                                                object : Radar.RadarBeaconCallback {
-                                                    override fun onComplete(
-                                                        status: Radar.RadarStatus,
-                                                        beacons: Array<RadarBeacon>?
-                                                    ) {
-                                                        if (status != Radar.RadarStatus.SUCCESS || beacons == null) {
-                                                            callTrackApi(null)
-
-                                                            return
-                                                        }
-
-                                                        callTrackApi(beacons)
-                                                    }
-                                                })
-                                        } else {
-                                            callTrackApi(arrayOf())
-                                        }
-                                    }
-                                },
-                                false
-                            )
-                        } else {
-                            callTrackApi(null)
-                        }
+                }
+
+                if (googlePlayProjectNumber == null) {
+                    Radar.logger.e("Error fetching Google Play project number")
+
+                    Radar.handler.post {
+                        callback?.onComplete(Radar.RadarStatus.ERROR_SERVER)
```

Comment:
Do we want to be early exiting here even if we have the project setting `Enable Android Play Integrity API: false`?

Reply:
Good callout

Reply:
There's maybe a scenario in which the Integrity API goes down or isn't working correctly for some subset of devices — I'd recommend instead that we just skip the  `verificationManager.getIntegrityToken` call and run the rest of `trackVerified` without it

Reply:
In the scenario where Google Play project number is `null` or otherwise not returned from the server, `verificationManager.getIntegrityToken()` will short circuit and return `(integrityToken = null, integrityException)`, and we'll proceed to call `POST /track`. So good to go.

In the scenario where the Integrity API goes down, we can handle that server-side. Can take that convo offline.